docs/source/explanation/algorithms/segmentation/semantic_segmentation.rst

```diff
@@ -1,23 +1,145 @@
++=======================+==============+============+=================+
+| Lite-HRNet-s-mod2     | N/A          | N/A        | N/A             |
++-----------------------+--------------+------------+-----------------+
+| Lite-HRNet-18         | N/A          | N/A        | N/A             |
++-----------------------+--------------+------------+-----------------+
+| Lite-HRNet-18-mod2    | N/A          | N/A        | N/A             |
++-----------------------+--------------+------------+-----------------+
+| Lite-HRNet-x-mod3     | N/A          | N/A        | N/A             |
++-----------------------+--------------+------------+-----------------+
 
 ************************
 Semi-supervised Learning
 ************************
 
-To be added soon
+To solve Semi-supervised learning problem for the semantic segmentation we use the `Mean Teacher algorithm <https://arxiv.org/abs/1703.01780>`_. The basic idea of this approach is to use two models during training: a "student" model, which is the main model being trained, and a "teacher" model, which acts as a guide for the student model.
+The student model is updated based on the ground truth annotations (for the labeled data) and pseudo-labels (for the unlabeled data) which are the predictions of the teacher model.
+The teacher model is updated based on the moving average of the student model's parameters. So, we don't use backward loss propagation for the teacher model's parameters.
+After training, only the student model is used for prediction.
+
+We utilize the same core algorithm's parameters as for the :ref:`supervised pipeline <semantic_segmentation_supervised_pipeline>`. The main difference is to use of different augmentation pipelines for the labeled and unlabeled data.
+We use only basic augmentations (random flip, random rotate, random crop) for the labeled data and more severe for the unlabeled (the same as we use for the :ref:`supervised pipeline <semantic_segmentation_supervised_pipeline>`).
+It helps with a better generalization and prevents unnecessary overfitting on the pseudo-labels generated by the teacher model.
+
+In the table below the `Dice score <https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient>`_ with our middle template on some datasets is presented. For comparison, we present the supervised baseline trained on the labeled data only.
+The results were obtained on our templates without any changes. We use 512x512 image resolution, for other hyperparameters, please, refer to the `related template <https://github.com/openvinotoolkit/training_extensions/blob/feature/otx/otx/algorithms/segmentation/configs/ocr_lite_hrnet_18_mod2/template.yaml>`_. We trained all models on 1 GPU Nvidia GeForce GTX3090.
+For `Cityscapes <https://www.cityscapes-dataset.com/>`_ and `Pascal-VOC <http://host.robots.ox.ac.uk/pascal/VOC/voc2007/index.html>`_ we use splits with different ratios of labeled to unlabeled data like `here <https://github.com/charlesCXK/TorchSemiSeg>`_.
```

Comment:
```suggestion
For `Cityscapes <https://www.cityscapes-dataset.com/>`_ and `Pascal-VOC <http://host.robots.ox.ac.uk/pascal/VOC/voc2012/index.html>`_ we use splits with different ratios of labeled to unlabeled data like `here <https://github.com/charlesCXK/TorchSemiSeg>`_.
```

Reply:
Is it more common to validate on the 2012 version?